spec/set-prototype-difference.html

```diff
@@ -4,14 +4,24 @@
     1. Let _set_ be the *this* value.
     1. If Type(_set_) is not Object, throw a *TypeError* exception.
     1. Let _Ctr_ be ? SpeciesConstructor(_set_, %Set%).
-    1. Let _newSet_ be ? Construct(_Ctr_, _set_).
-    1. Let _remover_ be ? Get(_newSet_, "delete").
-    1. If IsCallable(_remover_) is *false*, throw a *TypeError* exception.
-    1. Let _iteratorRecord_ be ? GetIterator(_iterable_).
+    1. Let _iteratorRecord_ be ? GetIterator(_set_).
+    1. If Type(_iterable_) is not Object, throw a *TypeError* exception.
+    1. Let _otherSet_ be _iterable_.
+    1. Let _hasCheck_ be ? Get(_otherSet_, "has").
+    1. If IsCallable(_hasCheck_) is *false*,
+        1. Let _otherSet_ be ? Construct(%Set%, _iterable_).
```

Comment:
?
```suggestion
        1. Let _otherSet_ be ? Construct(Ctr, _iterable_).
```

Reply:
k so if the argument wants to control how querying works, it’d have to provide a “has” method?

Reply:
Yup. (I'd prefer a symbol, but whatever.)